launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,294 @@
+
+      for (Map.Entry<String, String> entry : policy.getFields().entrySet()) {
+        String htmlFormPt2 = String.format("<input name='%s' value='%s' type='hidden' />\n", entry.getKey(), entry.getValue());
+        htmlForm.append(htmlFormPt2);
+      }
+      htmlForm.append(HTML_FORM_END);
+    } else {
+      htmlForm = new StringBuilder();
+      htmlForm.append(errorMessage);
+    }
+    return htmlForm.toString();
+  }
+
+  /**
+  * Display RSS feed xml string that user tries recalling with the given ID.
+  */
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+    String actionString = req.getParameter(ACTION);
+    String id = req.getParameter(ID);
+
+    if (actionString==null || id==null) {
+      res.setContentType("text/html");
+      res.getWriter().println("Please specify action and/or id.");
+      res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    Action act = Action.generateRSSLink; // initialization
+    Action action = act.valueOf(actionString);
+    
+    if (action == Action.generateRSSLink) {
+      // Generate link to the RSS feed
+      String rssLink = "https://launchpod-step18-2020.appspot.com/rss-feed?action=generateXml&id=" + id;
+      res.setContentType("text/html");
+      res.getWriter().println(rssLink);
+   } else if (action == Action.generateXml) {
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Key entityKey = null;
+      Entity desiredFeedEntity = null; 
+      try {
+        // Use key to retrieve entity from Datastore
+        entityKey = KeyFactory.stringToKey(id);
+        desiredFeedEntity = datastore.get(entityKey);
+
+      } catch (IllegalArgumentException e) {
+        // If entityId cannot be converted into a key
+        e.printStackTrace();
+        res.setContentType("text/html");
+        res.getWriter().println("Sorry, this is not a valid id.");
+        res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+        
+      } catch (EntityNotFoundException e) {
+        // No matching entity in Datastore
+        e.printStackTrace();
+        res.setContentType("text/html");
+        res.getWriter().println("Your entity could not be found.");
+        res.setStatus(HttpServletResponse.SC_NOT_FOUND);
+        return;
+      }
+
+      String podcastTitle = desiredFeedEntity.getProperty(PODCAST_TITLE).toString();
+      String description = desiredFeedEntity.getProperty(DESCRIPTION).toString();
+      String language = desiredFeedEntity.getProperty(LANGUAGE).toString();
+      String email = desiredFeedEntity.getProperty(EMAIL).toString();
+
+      EmbeddedEntity mp3Entity = (EmbeddedEntity) desiredFeedEntity.getProperty(MP3);
+      String mp3Link = mp3Entity.getProperty(MP3_LINK).toString();
+
+      RSS rssFeed = new RSS(podcastTitle, description, language, email, mp3Link);
+
+      XmlMapper xmlMapper = new XmlMapper();
```

Comment:
you shouldn't need to create an XmlMapper anywhere - use RSS#toXmlString whenever you need to get xml from an rss feed